src/withUser.js

```diff
@@ -8,11 +8,16 @@ export default Wrapped => {
       user: null
     };
 
+    handleError(error) {
+      this.setState({ user: null });
+      auth.logOut();
+    }
+
     componentDidMount() {
       fetch('/user', { headers: auth.header() })
         .then(res => res.json())
         .then(user => this.setState({ user }))
-        .catch(_ => this.props.history.push('/sign-in'));
```

Comment:
Should we still keep the redirect to sign-in?

If we need `history`, we can wrap `WithUser` with `withRouter`:
```js
import { withRouter } from 'react-router-dom';

export default Wrapped => {
  ...

  return withRouter(WithUser);
};
```